tests/test_covar2d.py

```diff
@@ -154,3 +156,59 @@ def testGetCWFCoeffsClean(self):
         self.assertTrue(
             np.allclose(results, self.coeff_cwf_clean, atol=utest_tolerance(self.dtype))
         )
+
+    def testGetCWFCoeffsCleanCTF(self):
+        """
+        Test case of clean images (coeff_clean and noise_var=0)
+        while using a non Identity CTF.
+
+        This case may come up when a developer switches between
+        clean and dirty images.
+        """
+
+        coeff_cwf = self.cov2d.get_cwf_coeffs(
+            self.coeff_clean, self.h_ctf_fb, self.h_idx, noise_var=0
+        )
+
+        # Reconstruct images from output of get_cwf_coeffs
+        img_est = self.basis.evaluate(coeff_cwf)
+        # Compare with clean images
+        delta = np.mean(np.square((self.imgs_clean - img_est).asnumpy()))
+        self.assertTrue(delta < 0.02)
```

Comment:
Why such a large error? Due to projection error from FB basis?

Reply:
No, I do not think it is from FB.  I think it is something with application and reversion of CTF or deeper... but fixing deeper issues is taking too long to be in the scope of this PR.

 I _think_ can use this as it is now.  It's up to you whether you want to stop other development to fix on cov2d more.

Reply:
Oh ok. If you can isolate a small example, might be worth making an issue.

Reply:
For sure, I'd like to fix this too, but I got distracted too long already. (I really thought this corner case was simpler...).

#424

Reply:
Thanks.